### DISCUSSION TITLE
Without Evil

### DISCUSSION THREAD

**Discussion by npajkovsky** [abougouffa/minemacs#155] [General]
Can it be used without evil? It seems that evil is very heavily backed into project.

**Comment by abougouffa** [comment]
Hello @npajkovsky 

Yes, it is possible to use the config without Evil, you should disable the `me-evil` module by removing it from the `minemacs-core-modules` variable in your [`~/.minemacs.d/modules.el` file](https://github.com/abougouffa/minemacs/blob/main/skel/modules.el).

However, all of my workflow depends heavily on Evil (the default keybindings, etc). But still, you can get a working MinEmacs without Evil just by disabling the `me-evil` module.

**Comment by stormwatch** [comment]
How did it go @npajkovsky? I am considering [Devil Mode](https://github.com/susam/devil)

**Comment by npajkovsky** [comment]
> How did it went @npajkovsky? I am considering [Devil Mode](https://github.com/susam/devil)

Not really well. `minemacs` worked very oddly, and I didn't have a time to dig in why. It's probably related to hooks.

Therefore, I'm still using Doom.

**Comment by almahdi** [comment]
magit breaks without evil mode, I didn't have time to look into the code. I just discovered the repo today.

Doom is so bloated, ended up writing my own config until I found this today.

**Reply by almahdi** [reply]
Okay, a quick fix for magit is defining a function:
```elisp
(defun evil-set-initial-state (a b) nil)
```

The problem in the config in modules/me-vc.el
```elisp
;; Enforce git commit conventions. See: chris.beams.io/posts/git-commit
(use-package git-commit
  :after magit
  :demand t
  :custom
  (git-commit-summary-max-length 72) ; defaults to Github's max commit message length
  (git-commit-style-convention-checks '(overlong-summary-line non-empty-second-line))
  :config
  (evil-set-initial-state 'git-commit-mode 'insert)
  (global-git-commit-mode 1))
```

**Reply by abougouffa** [reply]
Thanks for bringing this to my attention, I will wrap the `evil-set-initial-state` in a `with-eval-after-load` block.

**Comment by abougouffa** [comment]
Update:

I'm moving away from Evil these days, MinEmacs will be more focused on classical Emacs keybindings.

For these who want to use MinEmacs without Evil, you can just remove the `me-evil` from the list of modules in `minemacs-modules` (or the obsolete `minemacs-core-modules`). It should work without Evil now, without issues. I even moved all Evil/General custom keybindings to `me-evil` too.

**Reply by stormwatch** [reply]
Great work! Thank you so much.

I was wondering about what you found limiting while using evil. Is it that you wanted something more “modeless” or were there specific issues?
